### ISSUE TITLE
Multiple Linear Regression (left panel ONLY)

### ISSUE THREAD

**Issue by ashokkrish** [ashokkrish/CougarStats#5]
The “Multiple Linear Regression” option has been removed (hidden) from the radio button input. 

<img width="284" alt="image" src="https://github.com/ashokkrish/CougarStats/assets/410949/128a747a-16c7-4d41-a7ac-4d2d5194235e">

I would to bring back the radio button option called Multiple Linear Regression. When the user selects that option we ask them to upload a file

<img width="385" alt="image" src="https://github.com/ashokkrish/CougarStats/assets/410949/e17efe4a-fb47-4c56-bb70-5629c7b33b20">

Once the upload is complete the user gets to select one column as a the response variable (y) and multiple columns as the explanatory variables (x1, x2, x3, ...). 

<img width="376" alt="image" src="https://github.com/ashokkrish/CougarStats/assets/410949/8222360c-bbd5-4249-9aff-5b687f0792cf">

<img width="380" alt="image" src="https://github.com/ashokkrish/CougarStats/assets/410949/68b4089f-1133-43be-be91-cbf78221ead1">

In the above change the dropdown header to **Choose the Explanatory Variables (x1, x2, x3,...)**
also change the dropdown tooltip text to Select multiple variables

**Validations:**

- Make sure exactly one numeric column is selected for the response variable.
- Make sure one or more numeric columns are selected for the explanatory variables (the response variable column is now not selectable or hidden).

<img width="157" alt="image" src="https://github.com/ashokkrish/CougarStats/assets/410949/6ee45f95-f13c-4405-8e8a-95579885acb1">

The above should not appear for Multiple Linear Regression.

**Comment by samantha-v-brian** [comment]
I've begun this task and pushed what is functional so far. Please note that validation is not completely functional yet - still working on it. Let me know what you'd like to add/correct.

Also, could you please expand on this:
"Make sure one or more numeric columns are selected for the explanatory variables (the response variable column is now not selectable or hidden)." 
Does this mean the response variable column is not needed if explanatory variables are chosen?

Lastly, I've been using Simple Linear Regression files from the CougarStats Google Drive for testing purposes (ie. just to check if uploading and parsing file data is working). If there are specific Multiple Linear Regression files you'd like me to test with, especially for more error messages, please let me know.

Thanks!

**Comment by bryce-carson** [comment]
The following is taken from your email and an examination of the existing module.

# Multiple Linear Regression
- Pressing [Calculate] will result in a demand that data be available for use.
  ```R
  output$mainPanel <- renderUI{ validate(need(input$data, "Upload or enter useful data.")) }
  ```
- Validate that uploaded or manually entered data only contains numeric values; string values can be converted to factors. Non-numeric values will cause errors in `lm` object generation.
- When the user selects MLR they are able to upload a dataset.
- Before uploading [such a] file, if the user clicks "Calculate" then display a message on the mainPanel "Please upload a file."
- After the file upload is successful, the user [should] select a single response variable (denoted y).
- The user then selects one or more explanatory variables (denoted x1, x2, ...).
- Note if the [user] selects a single explanatory variable and clicks "Calculate" then they are simply running a Simple Linear Regression model.
- We could simply revert to the existing code. If the user indeed selects more than one explanatory variable then we are all set to fit a MLR.
- If everything looks fine then we create an `lm()` object in R,

```mermaid
flowchart TD
    A[Data Ingestion] -->|Validate is Numerical| B(Select Variables)
    B --> C{Multiple or Single Linear Regression}
    C -->|Single| D[Existing mainPanel]
    C -->|Multiple| E[TODO: mainPanel and server]
```

### Display/print on the main panel
Of the `lm` object:
  - the summary of that object
  - the extracted parameter estimates, and
  - the estimated regression

**Comment by ashokkrish** [comment]
@bryce-carson Forget to mention, please unhide the radio button toggle for SLR vs MLR up top.

**Comment by bryce-carson** [comment]
> @bryce-carson Forget to mention, please unhide the radio button toggle for SLR vs MLR up top.

No worries; I have a good idea of what to do here until we get to the statistical side.

**Comment by bryce-carson** [comment]
My immediate next steps, given what I've shown in #35, are to:
- Condition the scatterplot of the selected linear regression variables on simple linear regression versus multiple; the scatterplot will not be shown when multiple explanatory variables are selected.
- Convert the existing regression and correlation analysis (which is only effectively implemented for simple linear regression) into a Shiny module, so that I can show or hide the entire appropriate UI given the number of explanatory variables selected (as shown in the pseudo-code in a previous comment)

**Comment by ashokkrish** [comment]
@bryce-carson 

Let's have the top-level radioButton toggle for 

**Regression Type**
- [ ] Simple Linear Regression and Correlation Analysis
- [ ] Multiple Linear Regression

If the user selects  "Simple Linear Regression and Correlation Analysis" let's continue to allow for

<img width="383" alt="image" src="https://github.com/user-attachments/assets/1d2bb332-e261-4630-a409-738ff455e100" />

I want this back because majority of the users, at least for now, are students taking an Introductory Statistics course. Only a small fraction of students (for ex my Winter 2025 MATH 2444 class of students) would want the MLR option.

**Comment by bryce-carson** [comment]
# NOTE TO SELF: see the discussion in [CougarStats Tasks](https://docs.google.com/document/d/1jhT0I6mTvJbwVsYIcwk6YQweOdG0MRw6bVadqjYgtRU/edit?usp=sharing).

**Comment by ashokkrish** [comment]
@bryce-carson 

<img width="287" alt="image" src="https://github.com/user-attachments/assets/bec20b1e-b3bc-497e-b454-ddef6da0b6ef" />

Could you bold the text "**Regression Type**"? Perhaps

 `HTML("<label class='si-label'><b>Regression Type</b></label>")`

**Comment by ashokkrish** [comment]
@bryce-carson 

Under MLR I don't see

<img width="169" alt="image" src="https://github.com/user-attachments/assets/9dfbba3d-c095-4b04-bae3-5e302d559dea" />

Was it intentional? Are you thinking of running the lm() function and reactively display the results?

I would prefer if the user sees these two buttons after successfully importing the data. It is not always the case that they would run a full model. On the sidePanel there'll be checkBoxes for Tests and Plots to verify model assumptions.

**Comment by ashokkrish** [comment]
@bryce-carson 

Is it possible to move this from the mainPanel() to the sidePanel()?

<img width="544" alt="image" src="https://github.com/user-attachments/assets/372e025f-3be8-459c-983f-fa095f2b7782" />

I would like the mainPanel a clean slate to begin with. As soon as the data import is completed I want the data displayed on the mainPanel.

**Comment by bryce-carson** [comment]
> @bryce-carson
> 
> Is it possible to move this from the mainPanel() to the sidePanel()?
> <img alt="image" width="544" src="https://private-user-images.githubusercontent.com/410949/396301360-372e025f-3be8-459c-983f-fa095f2b7782.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQzOTU2MjgsIm5iZiI6MTczNDM5NTMyOCwicGF0aCI6Ii80MTA5NDkvMzk2MzAxMzYwLTM3MmUwMjVmLTNiZTgtNDU5Yy05ODNmLWZhMDk1ZjJiNzc4Mi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIxN1QwMDI4NDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xOTM1YmU3ZTZlYWFiYzIyZTdiNjRkYjUzMjBlNGQ3MjRiMTI2MWRiY2RjYTAzODU3NGE2YTg3YzFlMGRjOTNmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.-JGiTL_9M2yBwPEgcJmDPD1vZ2m21jvBAUiOsMgd3NE">
> 
> I would like the mainPanel a clean slate to begin with. As soon as the data import is completed I want the data displayed on the mainPanel.

It's possible, but this pictured module also supports the data _view_, so moving the import/view/update module to the side panel would defeat a data _view_ in the main panel when there is already a unified solution for this (which also ensures there is enough room for the module to look good). Having it in the side panel clutters things, I think.

Let's try keeping it in the main panel for now and see how it fits with other things once we've added those.

It is possible, though, yes. Moving the _view_ portion of the module to the main panel would not be possible (it's a module and I don't want to hack on something that is working nicely; I'd prefer to leave this as a black box). Producing _an additional_ view of the uploaded/imported data in the main panel would be redundant, so I think this module should remain in the main panel entirely. When the user has imported the data they can switch to the view tab of the module to view their data; that's the shortest, and already solved, solution to that request.

**Comment by bryce-carson** [comment]
> @bryce-carson
> 
> Under MLR I don't see
> <img alt="image" width="169" src="https://private-user-images.githubusercontent.com/410949/396300258-9dfbba3d-c095-4b04-bae3-5e302d559dea.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQzOTU2MjgsIm5iZiI6MTczNDM5NTMyOCwicGF0aCI6Ii80MTA5NDkvMzk2MzAwMjU4LTlkZmJiYTNkLWMwOTUtNGIwNC1iYWUzLTVlMzAyZDU1OWRlYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIxN1QwMDI4NDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yZDkzNjM2ODA2MzQxYWU3NTcyMmYzMDA0OWVlYzc2YWZlZmU5MWZhYTI5MmE5YWNkNzA3MzJiNjdlMzc5NThlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.ANmO4_8PJOM5d2DdoejaUJbHgpQKzUVT7TpVIxBtFoE">
> 
> Was it intentional? Are you thinking of running the lm() function and reactively display the results?
> 
> I would prefer if the user sees these two buttons after successfully importing the data. It is not always the case that they would run a full model. On the sidePanel there'll be checkBoxes for Tests and Plots to verify model assumptions.

So far everything is reactive, yes. Once data is imported and variables are selected the `lm()` function is immediately called with a new model. I can predicate that action on pressing the "Calculate" button, but one button that starts the engine is a little strange when there is (assumedly) going to be multiple steps to the process. Perhaps a "wizard" should be implemented in the main panel, and we can forgo the use of a "Calculate" button.

The "Reset" button was difficult to have, and therefore I removed it. I wasn't able to get the data upload to reset as nicely as I wanted. I will review this and expound upon this paragraph when I remember everything about it by trying it again.

**Comment by bryce-carson** [comment]
> @bryce-carson
> <img alt="image" width="287" src="https://private-user-images.githubusercontent.com/410949/396298321-bec20b1e-b3bc-497e-b454-ddef6da0b6ef.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzQzOTU2MjgsIm5iZiI6MTczNDM5NTMyOCwicGF0aCI6Ii80MTA5NDkvMzk2Mjk4MzIxLWJlYzIwYjFlLWIzYmMtNDk3ZS1iNDU0LWRkZWY2ZGEwYjZlZi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIxN1QwMDI4NDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00MDNiYWQxMTdiODBmYmU0YjhhYjU0Mjc5YWMwMzIzMGM5Nzg5NGM3OGJlNjYzYmQ5MDVmN2Y2ZjY0YTE2ZGRjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.fZEjZ4SCH2T-4SNlYZ__MlgvINmdcqo5pfHv4mgT914">
> 
> Could you bold the text "**Regression Type**"? Perhaps
> 
> `HTML("<label class='si-label'><b>Regression Type</b></label>")`

Absolutely.

**Comment by ashokkrish** [comment]
@bryce-carson Can you bold the following headers?

<img width="125" alt="image" src="https://github.com/user-attachments/assets/a97eadb1-18d7-42ea-9460-331006f678ca" />

<img width="230" alt="image" src="https://github.com/user-attachments/assets/876aee36-4a4c-4427-b21b-9bd46760209c" />

**Comment by ashokkrish** [comment]
Also got this error when I was playing around with the London Smoke data and tried swapping variables

```
Warning in model.matrix.default(mt, mf, contrasts) :
  the response appeared on the right-hand side and was dropped
Warning in model.matrix.default(mt, mf, contrasts) :
  problem with term 1 in model.matrix: no columns are assigned
Warning: Error in : value for ‘Smoke’ not found
  120: stop
  119: <Anonymous>
  118: mget
  112: eval [C:\Users\ashok\Desktop\CougarStatsApp\CougarStats\R\multipleLinearRegression.R#145]
  111: eval
  110: with.default
  108: eval [C:\Users\ashok\Desktop\CougarStatsApp\CougarStats\R\multipleLinearRegression.R#144]
  107: eval
  106: with.default
   97: renderUI [C:\Users\ashok\Desktop\CougarStatsApp\CougarStats\R\multipleLinearRegression.R#137]
   96: func
   83: renderFunc
   82: output$rc-MLR-linearModelEquations
    1: runApp
```

**Comment by bryce-carson** [comment]
> Got this error on the main panel
> 
> <img width="365" alt="image" src="https://github.com/user-attachments/assets/c4f1c153-2dad-43b5-983f-b1c5ca7d7b9d" />
> 
> With this on the RStudio console
> 
> ```
> Warning: Error in reformulate: 'termlabels' must be a character vector of length at least one
>   114: stop
>   113: reformulate
>   108: eval [C:\Users\ashok\Desktop\CougarStatsApp\CougarStats\R\multipleLinearRegression.R#142]
>   107: eval
>   106: with.default
>    97: renderUI [C:\Users\ashok\Desktop\CougarStatsApp\CougarStats\R\multipleLinearRegression.R#137]
>    96: func
>    83: renderFunc
>    82: output$rc-MLR-linearModelEquations
>     1: shiny::runApp
> ```

With what inputs?

**Comment by ashokkrish** [comment]
@bryce-carson 

- Move the ANOVA table to a new table panel labelled ANOVA

<img width="362" alt="image" src="https://github.com/user-attachments/assets/f7187690-9b24-401d-b5ab-45dfab03b315" />

- Is it possible to keep these tab panels hidden at first and show only after the import is successful and the user clicks **Calculate**?
<img width="427" alt="image" src="https://github.com/user-attachments/assets/696b34e1-b4f0-4035-8446-03d5b7d214f1" />

- When you run the MLR model for London Smoke data you get this
<img width="448" alt="image" src="https://github.com/user-attachments/assets/59c560b2-34af-4bb6-acf4-35ad57523aa2" />

Underneath this I would like to print the formula for Adjusted R^2 (capitalize R)
![image](https://github.com/user-attachments/assets/b4b502e4-e52d-45d0-8837-96b8e6d6f51c)

R^2 adj = 0.8355

Next print this standard interpretation statement (multiply R^2 Adj by 100):

**Interpretation:** Therefore, _____% of the variation is the response variable is explained by the multiple linear regression model - adjusted for the number of explanatory variables and sample size.

**Comment by bryce-carson** [comment]
> * Is it possible to keep these tab panels hidden at first and show only after the import is successful and the user clicks **Calculate**?

I have tried, but it's not working. The new tabPanel contents are being prioritized. See #45.

**Comment by bryce-carson** [comment]
> @bryce-carson
> 
>     * Move the ANOVA table to a new table panel labelled ANOVA
> 
> 
> <img alt="image" width="362" src="https://private-user-images.githubusercontent.com/410949/397248655-f7187690-9b24-401d-b5ab-45dfab03b315.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzUzMzgwNDcsIm5iZiI6MTczNTMzNzc0NywicGF0aCI6Ii80MTA5NDkvMzk3MjQ4NjU1LWY3MTg3NjkwLTliMjQtNDAxZC1iNWFiLTQ1ZGZhYjAzYjMxNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIyN1QyMjE1NDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zYjI1MjRjMDU4Y2FkOTQ2Y2NmZGNjYjQzOTg4NDhlODkxZTMxNTY1MWI5MWZjODFkYmI5NjNjY2E2YTg3ZTgxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.enDYg7yDSAdFySi0qmcQfY2eFLdbfNhRDBp6faGsUHU">
> 
>     * Is it possible to keep these tab panels hidden at first and show only after the import is successful and the user clicks **Calculate**?
> 
> 
> <img alt="image" width="427" src="https://private-user-images.githubusercontent.com/410949/397247314-696b34e1-b4f0-4035-8446-03d5b7d214f1.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzUzMzgwNDcsIm5iZiI6MTczNTMzNzc0NywicGF0aCI6Ii80MTA5NDkvMzk3MjQ3MzE0LTY5NmIzNGUxLWI0ZjAtNDAzNS04NDQ2LTAzZDViN2QyMTRmMS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIyN1QyMjE1NDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00OThiNTM5MmY5ZGU1OTI2N2M3N2VkN2NlOGJkZDNjMjY3YjczMTE3OTU1M2NhZGEyYTc3ZjA2ZGY2MjNlNTFiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.Tr1oPZT7-suvLtAL8t1AdTs9Yd7D2-aDxHg6z8RItsY">
> 
>     * When you run the MLR model for London Smoke data you get this
> 
> 
> <img alt="image" width="448" src="https://private-user-images.githubusercontent.com/410949/397247753-59c560b2-34af-4bb6-acf4-35ad57523aa2.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzUzMzgwNDcsIm5iZiI6MTczNTMzNzc0NywicGF0aCI6Ii80MTA5NDkvMzk3MjQ3NzUzLTU5YzU2MGIyLTM0YWYtNGJiNi1hY2Y0LTM1YWQ1NzUyM2FhMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIyN1QyMjE1NDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zZWRmM2ZiNjk0MDE5YTVmYzUyYzczNWUwYzEyN2M5ZjVhNGY5Yzk5ODg3MmI2MzA5ODViYzQ4OTA3YzU0YWVmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.87elosYKj3eJ11r_DEqBDXjrhNS4uPNLjoHG1siN7T8">
> 
> Underneath this I would like to print the formula for Adjusted R^2 (capitalize R) ![image](https://private-user-images.githubusercontent.com/410949/397246098-b4b502e4-e52d-45d0-8837-96b8e6d6f51c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzUzMzgwNDcsIm5iZiI6MTczNTMzNzc0NywicGF0aCI6Ii80MTA5NDkvMzk3MjQ2MDk4LWI0YjUwMmU0LWU1MmQtNDVkMC04ODM3LTk2YjhlNmQ2ZjUxYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIyN1QyMjE1NDdaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mYjc0NzBmMWRlYzAxY2IzNjFhZmI0YTViYWQxMjJjOTljYjQ5MzkxNmQ1YTllZjg2NjE0YjI0YmRhMTQyZTZkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.lQ0wtTdty3aUWruAYfOOANilTAi_3VsrnlyqpNamC6o)
> 
> R^2 adj = 0.8355
> 
> Next print this standard interpretation statement (multiply R^2 Adj by 100):
> 
> **Interpretation:** Therefore, _____% of the variation is the response variable is explained by the multiple linear regression model - adjusted for the number of explanatory variables and sample size.

@ashokkrish, in the example you show here there are four digits of precision in the summary of the model---as printed in the verbatim text---and then you keep all digits when you reformat as a percent (four digits, two following the decimal point). In my test with Example 5, Systolic Pressure, the verbatim text summary of the model has only three digits, and so I chose one digit of precision following the decimal point in the "____%" print.

@ashokkrish, how much precision do you want, always?

**Comment by ashokkrish** [comment]
@bryce-carson Let's go with 4 digits.

**Comment by ashokkrish** [comment]
@bryce-carson In SLR we were able to align the estimated regression equation to the left. Can you take a look at the relevant code?

For SLR can you make the two tabPanels look like how it is in MLR (I like the bold green **MLR, ANOVA, Diagnostics, Diagnostic Plots**)?

**Comment by bryce-carson** [comment]
> @bryce-carson In SLR we were able to align the estimated regression equation to the left. Can you take a look at the relevant code?

I will look and see how it was done, but I'm going to focus on the substantive things first.

> For SLR can you make the two tabPanels look like how it is in MLR (I like the bold green **MLR, ANOVA, Diagnostics, Diagnostic Plots**)?

I will see about that; it might require more changes than expected. Again, I will work on the substantive things in MLR first; if there is time remaining I will come back to this and other slight visual adjustments.

**Comment by bryce-carson** [comment]
> @bryce-carson In SLR we were able to align the estimated regression equation to the left. Can you take a look at the relevant code?

I looked and it does not use any _equation alignment environment_. If you prefer the equations be left-aligned _relative to the page_, but do not care if the equation elements themselves are aligned relative to each other (equals symbols, arithmetic symbols, variables, etc.) I can take achieve that.

**Comment by bryce-carson** [comment]
> @bryce-carson In SLR we were able to align the estimated regression equation to the left. Can you take a look at the relevant code?

I found a hack that works.

**Comment by bryce-carson** [comment]
`"p div.MathJax_Display".style("text-align: left;")`

The div element normally has the `text-align` property set to `center`.

**Comment by bryce-carson** [comment]
- [ ] Add a class to any paragraph or other tag containing MathJax display-style equations
- [ ] Use this class to select the child div with the MathJax_Display class
- [ ] Change the CSS of the finally selected tag

```js
$('p.MathJaxLeftAlign > div.MathJax_Display').css("text-align", "left")
```

**Comment by ashokkrish** [comment]
@bryce-carson Could you move the TS-value to this location?

<img width="352" alt="image" src="https://github.com/user-attachments/assets/d77fbe6d-eac2-45ab-802f-281fad43bc04" />

**Comment by bryce-carson** [comment]
> * [x]  Add a class to any paragraph or other tag containing MathJax display-style equations
> 
>     * [x]  Use this class to select the child div with the MathJax_Display class
> 
>     * [x]  Change the CSS of the finally selected tag
> 
> 
> ```js
> $('p.MathJaxLeftAlign > div.MathJax_Display').css("text-align", "left")
> ```

```R
JS(r"[ $('p.MathJaxLeftAlign > div.MathJax_Display').css("text-align", "left") ]")
```

This results in raw text being displayed; I imagine I just need to call the MathJax function in JavaScript.

**Comment by bryce-carson** [comment]
> @bryce-carson Could you move the TS-value to this location?
> <img alt="image" width="352" src="https://private-user-images.githubusercontent.com/410949/399121338-d77fbe6d-eac2-45ab-802f-281fad43bc04.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU0MjQyOTUsIm5iZiI6MTczNTQyMzk5NSwicGF0aCI6Ii80MTA5NDkvMzk5MTIxMzM4LWQ3N2ZiZTZkLWVhYzItNDVhYi04MDJmLTI4MWZhZDQzYmMwNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMjI4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTIyOFQyMjEzMTVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iNTM3MzljM2JhMTViNzJjMWFhMDViYjUyNWEwMWIwNjhkNmMyMmJkZjVhOWQ1NTgyM2I1ZWFiNWIyMDc5YThkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.GxsE06o46AT992DWfIjl_6jOeGHpWEdC6B0WMK8o0WY">
> 
> Further modifications
> 
>     * Bold the bottom row for Total

I'm not sure how to do that, but I can look into it if there's time.

>     * Change **p-Value of F** to **P-value**

OK.

>     * Header for the first column is **Source**
Those are row names, so there is no header; I can use a column with that header instead, however. It should display the same, mostly.

**Comment by bryce-carson** [comment]
- [ ] #46
- [ ] #47 
- [ ] #48

**Comment by bryce-carson** [comment]
@ashokkrish I have pushed a bunch of updates.

The MLR and ANOVA tab panels should be _mostly_ complete, apart from the second sentence of the interpretation of the p-value (comparing it against alpha), and the conclusion therefrom, on the ANOVA tab panel. The MLR tab panel should be 100% complete.

**Comment by bryce-carson** [comment]
@ashokkrish Please review those two tabs and let me know if there are any issues or content I didn't include, or content you'd like moved elsewhere.

I will finish the Diagnostics and Diagnostic Plots tabPanels tomorrow.

**Comment by ashokkrish** [comment]
@bryce-carson 

I would like 

<img width="231" alt="image" src="https://github.com/user-attachments/assets/0ae05532-0c07-410c-9d52-49f3949f18e6" />

to be in a column format with the following headers

Estimate Std. Error t value P-value 
<img width="301" alt="image" src="https://github.com/user-attachments/assets/f4dea260-8256-4da0-832e-b96b9d218417" />

**Comment by bryce-carson** [comment]
> @bryce-carson 
> 
> I would like 
> 
> <img width="231" alt="image" src="https://github.com/user-attachments/assets/0ae05532-0c07-410c-9d52-49f3949f18e6" />
> 
> to be in a column format with the following headers
> 
> Estimate Std. Error t value P-value 
> <img width="301" alt="image" src="https://github.com/user-attachments/assets/f4dea260-8256-4da0-832e-b96b9d218417" />
> 

Sure, I can do that.

When you wrote elsewhere that you wanted the coefficients I thought it was *just* the coefficients.

I'll include that table tomorrow.

**Comment by ashokkrish** [comment]
@bryce-carson 

Currently MLR for any dataset is resulting in the following error in the ANOVA tabPanel.

<img width="236" alt="image" src="https://github.com/user-attachments/assets/05828e4f-fccc-4a92-81c8-9869449969c9" />

**Comment by bryce-carson** [comment]
> @bryce-carson
> 
> I would like
> <img alt="image" width="231" src="https://private-user-images.githubusercontent.com/410949/400340694-0ae05532-0c07-410c-9d52-49f3949f18e6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY1NjUwOTUsIm5iZiI6MTczNjU2NDc5NSwicGF0aCI6Ii80MTA5NDkvNDAwMzQwNjk0LTBhZTA1NTMyLTBjMDctNDEwYy05ZDUyLTQ5ZjM5NDlmMThlNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDExMVQwMzA2MzVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lN2Y1M2ZjMjA5YjQxNzczNjk3ZTcwYjM0ZDE3ZDRkY2MxOTcwNGVlOTdmYTdiM2M3NjQ2MGQxYjY2NjlhNmQxJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.hnrIUBlM3mnmcxhg9c545OTM75ih_lQH70l2klcVu9A">
> 
> to be in a column format with the following headers
> 
> Estimate Std. Error t value P-value <img alt="image" width="301" src="https://private-user-images.githubusercontent.com/410949/400340599-f4dea260-8256-4da0-832e-b96b9d218417.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY1NjUwOTUsIm5iZiI6MTczNjU2NDc5NSwicGF0aCI6Ii80MTA5NDkvNDAwMzQwNTk5LWY0ZGVhMjYwLTgyNTYtNGRhMC04MzJlLWI5NmI5ZDIxODQxNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDExMVQwMzA2MzVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04YmRmMDI4MTk2YWEwN2UxM2I2OWUzMjRkNWRkN2FlNWFiNmQwZDEyNGM3OTFhMzVhMWNjMTIzYmFlMDJjNmUyJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.SbTZjVSj1DLIYq3DHbBXykJudUMND3-KtkIV54GFULo">

`summary(lm(hp ~ cyl + mpg + disp, data = mtcars))$coefficients` provides the table you're after. I'm including this now.

**Comment by bryce-carson** [comment]
> @bryce-carson
> 
> Currently MLR for any dataset is resulting in the following error in the ANOVA tabPanel.
> <img alt="image" width="236" src="https://private-user-images.githubusercontent.com/410949/401000198-05828e4f-fccc-4a92-81c8-9869449969c9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY1NjUwOTUsIm5iZiI6MTczNjU2NDc5NSwicGF0aCI6Ii80MTA5NDkvNDAxMDAwMTk4LTA1ODI4ZTRmLWZjY2MtNGE5Mi04MWM4LTk4Njk0NDk5NjljOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTExJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDExMVQwMzA2MzVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0xMjQxZjYxZmY0NmU5OGNkNTk0MjdjNTRiMTYwNTI4NGY2MGZhM2RjMzcyNjJiM2M1Y2UxNzkwYWE5NjRiYWM4JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.6jUp9GpeLWPMN31HlSKVz2L8NImn3ZyX9EDQR5mWF5M"> ```
> 
> Warning: Error in tibble::column_to_rownames: Can't find column `names` in `.data`. 109: 108: signalCondition 107: signal_abort 106: abort 105: tibble_abort 104: abort_unknown_column_names 103: tibble::column_to_rownames 102: %>% 101: renderTable [C:\Users\ashok\Desktop\CougarStatsApp\CougarStats\R\multipleLinearRegression.R#359] 100: func 87: renderFunc 86: output$rc-MLR-anovaTable 1: shiny::runApp

This was fixed in #60; you also mentioned this in #46.

**Comment by bryce-carson** [comment]
@ashokkrish Please try the application (not the `testMLRModule()` function) and give me a PASS or NO for proceeding with deployment to cougarstats.ca.

**Comment by ashokkrish** [comment]
@bryce-carson I uploaded the Example 1 - London_Smoke_MLR.xls data set and ran MLR.

**Request # 1** In the table could be report P-value to four significant digits please?

<img width="356" alt="image" src="https://github.com/user-attachments/assets/da5c5cf5-628e-479c-86ac-62737744f97f" />

There is an extra word Coefficients below the table that can be removed

<img width="216" alt="image" src="https://github.com/user-attachments/assets/7216f88a-06cd-4d85-a186-c5212f3bacc7" />

The extra words Confidence Intervals below the table can be removed as well.

**Request # 2** In the ANOVA tabPanel we have this 

<img width="377" alt="image" src="https://github.com/user-attachments/assets/e62b4ae5-003a-4db1-a300-a0ddd94ceb00" />

Could you add a column header for the first column as **Source**.

P-value in the last column can be rounded to four digits.

**Request # 3** Change the text **Test statistic** to **Test Statistic:**

Italicize and align center the column headers 
<img width="379" alt="image" src="https://github.com/user-attachments/assets/f748e79c-fe03-4cef-aa32-13335240f9bb" />

Drop the text **Using the P-Value method:** and replace it with **Conclusion:**

**Request # 4** In this expression

<img width="170" alt="image" src="https://github.com/user-attachments/assets/97984f26-97ad-4563-95d7-1a5eb7035e60" />
 include the actual R2 value ( = 0.8590).

You can drop this following expression from the output 
<img width="163" alt="image" src="https://github.com/user-attachments/assets/394110be-eb66-4cee-9f1f-0986b2efb86e" />

**Request # 5** IModify the following

<img width="319" alt="image" src="https://github.com/user-attachments/assets/da1e6ce6-80df-4c32-bade-83a20ab2891f" />

to

**Information Criteria**

Akaike Information Criterion (AIC): 166.3058

Bayesian Information Criterion (BIC): 169.1380

**Request # 6**  I am not sure if this dropdown is supposed to do something I am not seeing any plots of VIF values.

<img width="243" alt="image" src="https://github.com/user-attachments/assets/34cbaee9-208c-4216-96c8-167b8cce4c00" />

I am assuming this part of future work.

**Comment by ashokkrish** [comment]
@bryce-carson 

Just got this error on the RStudio console

```
Warning: Error in reformulate: 'termlabels' must be a character vector of length at least one
  111: stop
  110: reformulate
  105: eval [C:\Users\ashok\Desktop\CougarStatsApp\CougarStats\R\multipleLinearRegression.R#351]
  104: eval
  103: with.default
  101: renderUI [C:\Users\ashok\Desktop\CougarStatsApp\CougarStats\R\multipleLinearRegression.R#350]
  100: func
   87: renderFunc
   86: output$rc-MLR-anovaPValueMethod
    1: shiny::runApp
```

**Comment by bryce-carson** [comment]
> @bryce-carson 
> 
> Just got this error on the RStudio console
> 
> ```
> Warning: Error in reformulate: 'termlabels' must be a character vector of length at least one
>   111: stop
>   110: reformulate
>   105: eval [C:\Users\ashok\Desktop\CougarStatsApp\CougarStats\R\multipleLinearRegression.R#351]
>   104: eval
>   103: with.default
>   101: renderUI [C:\Users\ashok\Desktop\CougarStatsApp\CougarStats\R\multipleLinearRegression.R#350]
>   100: func
>    87: renderFunc
>    86: output$rc-MLR-anovaPValueMethod
>     1: shiny::runApp
> ```

Yes, this can safely be ignored. This only occurs when the Reset button is used, I believe.

If you restart the Shiny app and never use the reset button within MLR you shouldn't encounter it. It doesn't result in corruption output or break later states, so I'm okay with ignoring it.

**Comment by bryce-carson** [comment]
> **Request # 6**  I am not sure if this dropdown is supposed to do something I am not seeing any plots of VIF values.
> 
> <img width="243" alt="image" src="https://github.com/user-attachments/assets/34cbaee9-208c-4216-96c8-167b8cce4c00" />
> 
> I am assuming this part of future work.

There should be either a warning that VIFs couldn't be calculated, or a table of the VIFs in the model.

I will try it with the example data you mentioned.

**Comment by bryce-carson** [comment]
> P-value in the last column can be rounded to four digits.

I was able to increase the number of _significant digits_ for _just this column_ to four, but that includes a lot of _leading zeros_. To have the number of _decimal places_ always be four _for just this column_ I am unsure how to do that.

Is the first solution acceptable? `0.00000004321` is a number you might see, along with `0.000001`, or `0.1234`, or `0.1`, but `0.1000` wouldn't display, and `0.0000` wouldn't display.


@ashokkrish

**Comment by bryce-carson** [comment]
> > **Request # 6**  I am not sure if this dropdown is supposed to do something I am not seeing any plots of VIF values.
> > <img alt="image" width="243" src="https://private-user-images.githubusercontent.com/410949/402806303-34cbaee9-208c-4216-96c8-167b8cce4c00.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzY5OTQ0MDgsIm5iZiI6MTczNjk5NDEwOCwicGF0aCI6Ii80MTA5NDkvNDAyODA2MzAzLTM0Y2JhZWU5LTIwOGMtNDIxNi05NmM4LTE2N2I4Y2NlNGMwMC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTE2JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDExNlQwMjIxNDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jYzQyOWJlYmNmMmZjMDViYWFkNGQzZDI2ZjE1ZjRmMmU3OWZmMDdiMDdlODBjZjIxNzc3YjVhZmUyZWY3MTdlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.r4ld9mlTQDQofz8cxygDUQ9CSJd4XhN6NMtMBDHaPdo">
> > I am assuming this part of future work.
> 
> There should be either a warning that VIFs couldn't be calculated, or a table of the VIFs in the model.
> 
> I will try it with the example data you mentioned.

@ashokkrish, I have discovered that the implementation I made for this only works in the testing applications, and oddly does not work when the MLR module is embedded in the larger CougarStats application. I will need to do more testing to figure out why that is, and then rectify that issue.

**Comment by bryce-carson** [comment]
@ashokkrish I have resolved that bug which caused the Graphical and VIFs Detection Methods to not appear in the full application. There is a new visual bug, however, at least in Safari, wherein the footer displays over (occluding) the graphical plot. It doesn't affect /small VIFs tables/, however.

I will look into the graphical issue tomorrow for a short while. Please consider this a back burner task, however. The plot may always be viewed in full by right clicking the plot and selecting "Open image in new tab" as a work-around for now.

Please review the rest of the MLR module once more and let me know if I can go ahead and push the updates to the CougarStats.ca domain.

**Comment by ashokkrish** [comment]
@bryce-carson 

The degrees of freedom for an ANOVA table is always an integer. Could we drop the leading zeroes?

<img width="398" alt="image" src="https://github.com/user-attachments/assets/b72c19b9-a823-415a-a145-5a215e4fe680" />

df would simply be 2 12 and 14 in the above table for the London Smoke dataset.

**Comment by bryce-carson** [comment]
> @bryce-carson 
> 
> The degrees of freedom for an ANOVA table is always an integer. Could we drop the leading zeroes?
> 
> <img width="398" alt="image" src="https://github.com/user-attachments/assets/b72c19b9-a823-415a-a145-5a215e4fe680" />
> 
> df would simply be 2 12 and 14 in the above table for the London Smoke dataset.
> 

I believe so. I just need to change the class of the number to integer from numeric.

**Comment by ashokkrish** [comment]
@bryce-carson Change the button label from Reset to Reset Values as this consistent with every other place.

**Comment by bryce-carson** [comment]
@ashokkrish Both of those requests are completed. Let me know if I can deploy this to CougarStats.ca or if there's anything else.

**Comment by ashokkrish** [comment]
> [@ashokkrish](https://github.com/ashokkrish) Both of those requests are completed. Let me know if I can deploy this to CougarStats.ca or if there's anything else.

@bryce-carson Let's wait for another week for a deployment. I have some minor changes to the ANOVA and Diagnostics tabPanels and pushed it to the main branch. Please pull them.

**Comment by ashokkrish** [comment]
I would like to see a similar Correlation Matrix

<img width="464" alt="Image" src="https://github.com/user-attachments/assets/19fdbbc6-9b9e-4d36-8e4c-89d9fd031d41" />

**Comment by bryce-carson** [comment]
> I would like to see a similar Correlation Matrix
> <img alt="Image" width="464" src="https://private-user-images.githubusercontent.com/410949/404605313-19fdbbc6-9b9e-4d36-8e4c-89d9fd031d41.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzcyNjE1MjUsIm5iZiI6MTczNzI2MTIyNSwicGF0aCI6Ii80MTA5NDkvNDA0NjA1MzEzLTE5ZmRiYmM2LTliOWUtNGQzNi04ZTRjLTg5ZDlmZDAzMWQ0MS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTE5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDExOVQwNDMzNDVaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lMTExZDEyNTVjMjJkNmU3OWM5ODk5MTYzZGVmNDBlNzYzZWE0ZTY0MTA2NDQwODY1YzBjZTFlOTlkN2RkZTM0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.509qdnk-58W9t_kYJbHpmnCBoxFdqrvRhKl-Tv1Z2s4">

I will look at extensions to ggplot2, or the options of DT, to see if either of these has support built-in for display of matrices in this manner. If not, I can definitely write a small algorithm to change elements of the table above the matrix diagonal to `NA` or `""` so that they are not rendered by DT.

**Comment by ashokkrish** [comment]
I believe for a dataset set, for example, with k = 4, the following code prints the correlation matrix

 `cor(data[,c("x1","x2","x3","x4")], use ="complete")` 

Can you check? A lower triangular matrix is ideal. In the future I would like to bold or highlight those correlation values that are, say, below -0.75 or above +0.75 because they convey the variables are **highly correlated** with a recommendation that one of them be dropped for a better fit.

**Comment by bryce-carson** [comment]
> I believe for a dataset set, for example, with k = 4, the following code prints the correlation matrix
> 
> `cor(data[,c("x1","x2","x3","x4")], use ="complete")`
> 
> Can you check? A lower triangular matrix is ideal.

```R
library(tidyverse)
cor(mtcars, use = "complete.obs")
```

This indeed prints a correlation matrix.

**Comment by bryce-carson** [comment]
> In the future I would like to bold or highlight those correlation values that are, say, below -0.75 or above +0.75 because they convey the variables are **highly correlated** with a recommendation that one of them be dropped for a better fit.

Requests like this are great. They provide a long-term goal and a design which can be worked towards.

If you look at section 2.10 of the DT manual online you'll see that HTML can be escaped. We will choose to _not_ escape HTML content and I'll hack together something that will generate the HTML for rendering (like any other Shiny render function).

DT doesn't support such table rendering options out of the box (hiding the elements above or below the diagonal of a matrix), but I can write a function which wraps this behaviour and customization options so that you can use it elsewhere if you like it.

See [the documentation of the functions in the DT package](https://rstudio.github.io/DT/functions.html) (which we're using already) to see what is possible for future inspiration.

As part of the six hours you requested I work (which I'll do tomorrow, according my plans right now) I'll hack this solution together so that you can have this now, rather than later.

**Comment by ashokkrish** [comment]
> library(tidyverse)
> cor(mtcars, use = "complete.obs")
> This indeed prints a correlation matrix.

Leaving [this](https://stackoverflow.com/questions/44785229/change-chart-correlation-defaults-to-produce-best-fit-line-rather-than-smoothed) StackOverflow link here for future reference.

**Comment by ashokkrish** [comment]
**FUTURE WORK:** The user/instructor may want to upload a data from an existing R package. For example, the `stats` package contains several good datasets. 

<img width="574" alt="Image" src="https://github.com/user-attachments/assets/c6e0ad6c-fe49-4865-866a-ee821805c5a9" />

Rather than uploading the bikes data from a local file why not import the data from the stats package by specifying the library name and dataset name.

**Comment by bryce-carson** [comment]
> **FUTURE WORK:** The user/instructor may want to upload a data from an existing R package. For example, the `stats` package contains several good datasets. 
> 
> <img width="574" alt="Image" src="https://github.com/user-attachments/assets/c6e0ad6c-fe49-4865-866a-ee821805c5a9" />
> 
> Rather than uploading the bikes data from a local file why not import the data from the stats package by specifying the library name and dataset name.
> 
> 

That's a feature of the data import module. When initializing the module you are able to state several options, those being Uploaded Dat, Input Data (such as in text), or Environment Data (such as from a package).

I'll upload a video later.

**Comment by bryce-carson** [comment]
> **FUTURE WORK:** The user/instructor may want to upload a data from an existing R package. For example, the `stats` package contains several good datasets.
> <img alt="Image" width="574" src="https://private-user-images.githubusercontent.com/410949/406631844-c6e0ad6c-fe49-4865-866a-ee821805c5a9.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzc4MzY5NjgsIm5iZiI6MTczNzgzNjY2OCwicGF0aCI6Ii80MTA5NDkvNDA2NjMxODQ0LWM2ZTBhZDZjLWZlNDktNDg2NS04NjZhLWVlODIxODA1YzVhOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMTI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDEyNVQyMDI0MjhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01YmVkNjVkOWJkNWYyMzNkZDkxNjgzMTQ3OWJjNDY0MGNlODIyN2MzZDc5ODIyYzA4YzdmOGMzMjY2NDE4MWQ3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.NdDqRd18WO0NqouD9mkEj0jpHWleE8WCIXCz-OeblmU">
> 
> Rather than uploading the bikes data from a local file why not import the data from the stats package by specifying the library name and dataset name.

Watch this short video to see how to select a data frame from an environment (in our case, one of the many packages we have loaded). We get this functionality "for free" from the Data Import module.

https://github.com/user-attachments/assets/99e95f4e-9618-4993-b403-99188e414afa

**Comment by bryce-carson** [comment]
@ashokkrish, the correlation table is now like you ask (except for bold/highlighting). I haven't been able to figure out why the plotOutput isn't rendering in a normal flow, except that what I'm doing "feels exotic." I have multiple nestings of renderUI and it doesn't seem to be worthwhile, when in other areas of the application we don't offer a drop-down choice. I'm going to reel it back and just let the page be rendered correctly, if not as pretty, by listing multiple methods at once (rather than letting a user "choose" a correlation detection method with a dropdown, they can "choose" one by reading that? 🤷🏼).

**Comment by ashokkrish** [comment]
> I'm going to reel it back and just let the page be rendered correctly, if not as pretty, by listing multiple methods at once (rather than letting a user "choose" a correlation detection method with a dropdown, they can "choose" one by reading that?

This is good. Please proceed as you suggest.

**Comment by ashokkrish** [comment]
@bryce-carson There appears to be a duplication of plots rendered in the **Diagnostic Plots** tabPanel. For a particular exercise I have the following as the first plot

![Image](https://github.com/user-attachments/assets/b1ec53ec-938f-4d83-8b1c-5671dadc8b0a)

This exact sample plot is duplicate of one of the 2 x 2 panel of plots printed right below it.

![Image](https://github.com/user-attachments/assets/61c1d6e8-20c8-4eb1-9735-0fd85ecb6e49)

I suggest we hide the top plot (comment the code for now) and only render the 2 x 2 panel of plots.

**Comment by ashokkrish** [comment]
In the Multicollinearity the Correlation Matrix is not showing up properly.

**Comment by bryce-carson** [comment]
> @bryce-carson In the **Multicollinearity Detection** tabPanel the _Correlation Matrix_ is not showing up properly. 

Thanks. I know. I'm still working out the final issues before I move on to the p-values. It will be tight but I think I can get everything done and release by 31st.
